lib/CAPI/Firtool/Firtool.cpp

```diff
@@ -240,48 +287,48 @@ DEFINE_FIRTOOL_OPTION_BOOL(StripDebugInfo, stripDebugInfo)
 // Populate API.
 //===----------------------------------------------------------------------===//
 
-MlirLogicalResult firtoolPopulatePreprocessTransforms(MlirPassManager pm,
-                                                      FirtoolOptions options) {
+MlirLogicalResult firtoolPopulatePreprocessTransforms(
+    MlirPassManager pm, FirtoolPreprocessTransformsOptions options) {
   return wrap(
       firtool::populatePreprocessTransforms(*unwrap(pm), *unwrap(options)));
 }
 
-MlirLogicalResult
-firtoolPopulateCHIRRTLToLowFIRRTL(MlirPassManager pm, FirtoolOptions options,
-                                  MlirModule module,
-                                  MlirStringRef inputFilename) {
+MlirLogicalResult firtoolPopulateCHIRRTLToLowFIRRTL(
+    MlirPassManager pm, FirtoolCHIRRTLToLowFIRRTLOptions options,
+    MlirModule module, MlirStringRef inputFilename) {
```

Comment:
move `MlirModule module, MlirStringRef inputFilename` to `options`?

Reply:
Move `MlirStringRef inputFilename` should be fine. How about keep `MlirModule module`? IMO it's more like a parameter instead of a config-option.

Reply:
sounds reasonable, but why does a `MlirModule` is required here?

Reply:
Removed both `module` and `inputFilename`. Setting `blackBoxRootPath` from Firtool-bin if it's empty, so `inputFilename` is no longer needed in Firtool-lib.